packages/core/frontend/sirius-components-core/src/workbench/WorkbenchMachine.ts

```diff
@@ -117,6 +120,7 @@ export const workbenchMachine = Machine<WorkbenchContext, WorkbenchStateSchema,
               UPDATE_SELECTION: {
                 target: 'initial',
                 actions: 'updateSelection',
+                cond: (context, event) => !equals(context.selection, event.selection),
```

Comment:
I'll move this code to the guard section of the state machine, it's easier to read than just inline like that.